ote_cli/ote_cli/tools/explain.py

```diff
@@ -0,0 +1,168 @@
+    parser.add_argument("template")
+    parser.add_argument(
+        "-i",
+        "--input",
+        required=True,
+        help="Source of input data: images folder, image, webcam and video.",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        default="saliency_dump",
+        help="Output path for explanation images.",
+    )
+    parser.add_argument(
+        "--load-weights",
+        required=True,
+        help="Load only weights from previously saved checkpoint",
+    )
+    parser.add_argument(
+        "--explain-model",
+        default="EigenCAM",
+        help=f"XAI model name, currently support {SUPPORTED_XAI_MODELS}",
+    )
+    add_hyper_parameters_sub_parser(parser, hyper_parameters, modes=("INFERENCE",))
+
+    return parser.parse_args(), template, hyper_parameters
+
+
+def get_explain_map(task, frame):
+    """
+    Returns list of explanations made by task on frame and time spent on doing explanation.
+    """
+
+    empty_annotation = AnnotationSceneEntity(
+        annotations=[], kind=AnnotationSceneKind.PREDICTION
+    )
+
+    item = DatasetItemEntity(
+        media=Image(cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)),
+        annotation_scene=empty_annotation,
+    )
+
+    dataset = DatasetEntity(items=[item])
+
+    start_time = time.perf_counter()
+    explain_map = task.explain(
+        dataset,
+        InferenceParameters(is_evaluation=True),
+    )
+    elapsed_time = time.perf_counter() - start_time
+    saliency_map = explain_map[0]
+    return saliency_map, elapsed_time
+
+
+def main():
+    """
+    Main function that is used for model demonstration.
+    """
+
+    # Dynamically create an argument parser based on override parameters.
+    args, template, hyper_parameters = parse_args()
+    # Get new values from user's input.
+    updated_hyper_parameters = gen_params_dict_from_args(args)
+    # Override overridden parameters by user's values.
+    override_parameters(updated_hyper_parameters, hyper_parameters)
+
+    hyper_parameters = create(hyper_parameters)
+
+    # Get classes for Task, ConfigurableParameters and Dataset.
+    if args.load_weights.endswith(".pth"):
+        task_class = get_impl_class(template.entrypoints.base)
+    else:
+        raise ValueError(f"Unsupported file: {args.load_weights}.")
+
+    environment = TaskEnvironment(
+        model=None,
+        hyper_parameters=hyper_parameters,
+        label_schema=read_label_schema(args.load_weights),
+        model_template=template,
+    )
+
+    environment.model = read_model(
+        environment.get_model_configuration(), args.load_weights, None
+    )
+
+    task = task_class(task_environment=environment)
+
+    imgs = get_image_files(args.input)
+    if imgs is None:
+        raise ValueError(f"No image found in {args.input}")
+
+    if not os.path.exists(args.output):
+        os.makedirs(args.output)
+
+    elapsed_times = deque(maxlen=10)
+
+    for _, (root_dir, filename) in enumerate(imgs):
```

Comment:
Right. Since the current form only considers the small image input, the data parsing part need to be fixed.

Reply:
Fixed at commit [e58215d](https://github.com/openvinotoolkit/training_extensions/pull/1311/commits/e58215d65fd5558b47b93af178635d3aee767566), to get images using `dataset_class`. And we call `ClsExplainer` in `task.explain`, so that we can use batch processing.